### ISSUE TITLE
apply.sh: Failed to connect to localhost port 8111: Connection refused

### ISSUE THREAD

**Issue by flaccid** [rancher/rancher#3668]
Version:
v0.59.0

Steps: 
We upgraded to v59 (I believe from 56 or 57) and are seeing the following in the server log. It appears to be running a script that tries to connect to localhost. Wondering if this is an issue, how to solve it etc.?

One thing to note is that the Rancher server IP was changed.

Results: 
```
[[scripts.sh:184] script_env
[[scripts.sh:109] '[' -n http://rancher.auiag.corp/v1 ']'
[[scripts.sh:110] return
[[scripts.sh:186] export CATTLE_HOME=/var/lib/cattle
[[scripts.sh:186] CATTLE_HOME=/var/lib/cattle
[[scripts.sh:187] export CATTLE_CONFIG_URL=http://rancher.auiag.corp/v1
[[scripts.sh:187] CATTLE_CONFIG_URL=http://rancher.auiag.corp/v1
[[scripts.sh:188] export CATTLE_STORAGE_URL=http://rancher.auiag.corp/v1
[[scripts.sh:188] CATTLE_STORAGE_URL=http://rancher.auiag.corp/v1
[apply.sh:5] stage_files
[scripts.sh:77] local monit=false
[scripts.sh:79] '[' -d content-home ']'
[scripts.sh:80] find content-home -name '*.sh' -exec chmod +x '{}' ';'
[scripts.sh:81] sed 's/^/INFO: HOME -> /'
[scripts.sh:81] tar cf - -C content-home .
[scripts.sh:81] tar xvf - --no-overwrite-dir -C /var/lib/cattle
INFO: HOME -> ./
INFO: HOME -> ./etc/
INFO: HOME -> ./etc/cattle/
INFO: HOME -> ./etc/cattle/ipsec/
INFO: HOME -> ./etc/cattle/ipsec/config.json
[scripts.sh:84] '[' -d content ']'
[scripts.sh:102] '[' false = true ']'
[apply.sh:7] '[' '' '!=' true ']'
[apply.sh:8] curl -f http://localhost:8111/v1/reload
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8111: Connection refused
[apply.sh:9] echo Failed to reload ipsec config
Failed to reload ipsec config
[apply.sh:10] curl http://localhost:8111/v1/reload '{'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8111: Connection refused
curl: (3) [globbing] unmatched brace in column 1
]
2016-02-22 05:12:34,887 ERROR [:] [] [] [] [ecutorService-4] [.p.c.v.i.ConfigItemStatusManagerImpl] Failed null, exit code [3] output [INFO: Updating ipsec-hosts
```

Expected: 'no errors'

**Comment by deniseschannon** [comment]
When did the Rancher server IP change? At one point during the upgrade?

**Comment by flaccid** [comment]
Hi @deniseschannon this was a export of the mysql from the rancher container, then stop rancher. We put the db on its own server, then ran up a new rancher container with v59, the host for this server container is on a different IP to before. Not sure if this would have anything to do with it, but thought I should mention. Looking at the log this may not be happening any longer (we also replaced most of the hosts in the environment now).

**Comment by ibuildthecloud** [comment]
The best way to debug networking issue is with the `/var/log/rancher-net.log` and the `/var/log/charon.log`.  We use `monit` to keep things running, lately we've been seeing some oddness in which monit thinks something is running and it's not.  Can you log into the network agent and run `monit status` and see if `rancher-net` is running.  `rancher-net` is what listens to port 8111.

**Comment by ibuildthecloud** [comment]
Also look at `ps -ef` and see if it is really running.

**Comment by cedvan** [comment]
Hi,

I have same problem, `monit status` out :
```
Process 'rancher-net'
  status                            Does not exist
  monitoring status                 Monitored
  data collected                    Mon, 07 Mar 2016 20:53:19
```
A solution ?

**Comment by cedvan** [comment]
```
$ sudo service rancher-net start
Cannot open netlink socket: Protocol not supported
ERRO[0001] Failed to connect to charon: dial unix /var/run/charon.vici: connect: connection refused 
ERRO[0002] Failed to connect to charon: dial unix /var/run/charon.vici: connect: connection refused 
2016/03/07 20:57:22 Failed to talk to charon:%!(EXTRA *net.OpError=dial unix /var/run/charon.vici: connect: connection refused)
charon: no process found
 * Starting rancher-net configuration rancher-net
   ...done.
```
```
$ sudo service charon start
 * Starting charon configuration charon
   ...done.
```

charon stay Execution failed :
```
$ tail -f /var/log/charon.log
Mar  7 21:00:23 00[DMN] Starting IKE charon daemon (strongSwan 5.3.5, Linux 4.4.4-docker-root, x86_64)
Mar  7 21:00:23 00[KNL] unable to create netlink socket
Mar  7 21:00:23 00[NET] installing IKE bypass policy failed
Mar  7 21:00:23 00[NET] installing IKE bypass policy failed
Mar  7 21:00:23 00[NET] enabling UDP decapsulation for IPv6 on port 4500 failed
Mar  7 21:00:23 00[NET] installing IKE bypass policy failed
Mar  7 21:00:23 00[NET] installing IKE bypass policy failed
Mar  7 21:00:23 00[NET] enabling UDP decapsulation for IPv4 on port 4500 failed
Mar  7 21:00:23 00[LIB] feature CUSTOM:libcharon in critical plugin 'charon' has unmet dependency: CUSTOM:kernel-ipsec
Mar  7 21:00:23 00[CFG] loading ca certificates from '/usr/local/etc/ipsec.d/cacerts'
Mar  7 21:00:23 00[LIB] opening directory '/usr/local/etc/ipsec.d/cacerts' failed: No such file or directory
Mar  7 21:00:23 00[CFG]   reading directory failed
Mar  7 21:00:23 00[CFG] loading aa certificates from '/usr/local/etc/ipsec.d/aacerts'
Mar  7 21:00:23 00[LIB] opening directory '/usr/local/etc/ipsec.d/aacerts' failed: No such file or directory
Mar  7 21:00:23 00[CFG]   reading directory failed
Mar  7 21:00:23 00[CFG] loading ocsp signer certificates from '/usr/local/etc/ipsec.d/ocspcerts'
Mar  7 21:00:23 00[LIB] opening directory '/usr/local/etc/ipsec.d/ocspcerts' failed: No such file or directory
Mar  7 21:00:23 00[CFG]   reading directory failed
Mar  7 21:00:23 00[CFG] loading attribute certificates from '/usr/local/etc/ipsec.d/acerts'
Mar  7 21:00:23 00[LIB] opening directory '/usr/local/etc/ipsec.d/acerts' failed: No such file or directory
Mar  7 21:00:23 00[CFG]   reading directory failed
Mar  7 21:00:23 00[CFG] loading crls from '/usr/local/etc/ipsec.d/crls'
Mar  7 21:00:23 00[LIB] opening directory '/usr/local/etc/ipsec.d/crls' failed: No such file or directory
Mar  7 21:00:23 00[CFG]   reading directory failed
Mar  7 21:00:23 00[CFG] loading secrets from '/usr/local/etc/ipsec.secrets'
Mar  7 21:00:23 00[CFG] opening secrets file '/usr/local/etc/ipsec.secrets' failed: No such file or directory
Mar  7 21:00:23 00[LIB] failed to load 1 critical plugin feature
Mar  7 21:00:23 00[DMN] initialization failed - aborting charon
```

**Comment by cedvan** [comment]
Files and folder missing, possible to regenerate ? From where comes this problem ?

**Comment by cedvan** [comment]
I have this problem with upgrade to 0.59.1. I didn't this problem before with 0.56.1. But now, after downgrade, problem is present with 0.56.1... Why ?

**Comment by flaccid** [comment]
Apologies @ibuildthecloud I didn't have time to check with existing agent, we've replaced the cows alot. Will look out for it again, @cedvan info looks quite informative to help.

**Comment by cedvan** [comment]
@flaccid So, do you can help me ?

**Comment by cedvan** [comment]
@flaccid : I use [docker-root](https://github.com/ailispaw/docker-root) vagrant box because I want static ip on network LAN. RancherOS with vagrant declare an ip by DHCP... With rancherOS, charon works. But with docker-root, folder and files are missing in agent-instance to launch charon. how I can solve it please ?

**Comment by flaccid** [comment]
@cedvan if not specifically the problem in this issue probably best to raise a separate one or post on the forums. I'm just the guy that reported the original issue.

**Comment by ibuildthecloud** [comment]
@cedvan. You kernel doesn't support IPSec.  What host OS are you running

**Comment by ibuildthecloud** [comment]
@flaccid I think the issue here is that /var/run in the network agent is not deleted on restart and that messes up the PID files.  I'm going to change the way we use monit to not use pid files.  Also cleanup that folder.

**Comment by flaccid** [comment]
Sounds good to me ☺

**Comment by cedvan** [comment]
@ibuildthecloud, Ok, I use [docker-root](https://github.com/ailispaw/docker-root) as host. I'll see to implement ipsec support. Thank you !

**Comment by cedvan** [comment]
@ibuildthecloud, [docker-root](https://github.com/ailispaw/docker-root) have Linux kernel v4.4.6, and Rancher OS v4.2. I think v4.4.6 as the IPSec support, no ? The problem must be elsewhere...

**Comment by ibuildthecloud** [comment]
@cedvan The kernel supports it, but depends on whether the right options are enabled.  I just looked at https://github.com/ailispaw/docker-root/blob/master/configs/kernel.config and it doesn't seem to have ipsec.  Here's a list of the required options:

https://wiki.strongswan.org/projects/strongswan/wiki/KernelModules

**Comment by cedvan** [comment]
@ibuildthecloud, Ok it's right, problem solved !

**Comment by ibuildthecloud** [comment]
On boot we clear `/var/run/*.pid` now and also we have switch to regexp based monitoring of processes.  https://github.com/rancher/cattle/pull/1526

**Comment by deniseschannon** [comment]
@flaccid @cedvan This was addressed in our v1.0.0-rcX releases. Currently, we don't support upgrades to those releases yet, but when we do, if you could upgrade to one and see if you still have this issue, please let me know.

**Comment by cedvan** [comment]
It's solved for me :wink: ! Thank you
